### DISCUSSION TITLE
GLAuth as an LDAP auth backend?

### DISCUSSION THREAD

**Discussion by prologic** [authelia/authelia#4234] [Q&A]
Hi all 👋

Wondering if anyone has tried and if so had any luck integrating GLAuth as an LDAP backend for Authelia?

- [glauth/glauth: A lightweight LDAP server for development, home use, or CI](https://github.com/glauth/glauth)

Kind regards

James

**Comment by james-d-elliott** [comment]
I think someone did try it. I think there were a few nuances but it worked. Other users have used lldap as well. If you wanted to get it working I'd be more than willing to help set it up, and even create implementation defaults for it: https://www.authelia.com/reference/guides/ldap/#attribute-defaults

**Reply by james-d-elliott** [reply]
I think GLAuth requires this, but wait until you get an error about the RootDSE search before enabling (only available in 4.37): https://www.authelia.com/configuration/first-factor/ldap/#permit_feature_detection_failure

Also note there are two chat options available at the bottom when creating new issues. Matrix/Discord.

**Reply by prologic** [reply]
Thanks! I'm _actually_ rethinking what I want to do here... I'm thinking instead of LDAP (only because that's the only other option Authelia supports right now) I instead build an [IndieAuth](https://indieauth.com/) implementation (assuming there are flexible interfaces I can build against?)

> Also note there are two chat options available at the bottom when creating new issues. Matrix/Discord.

So sorry, I use neither of these. Is IRC an option? 🤔

**Reply by prologic** [reply]
To elaborate slightly, part of my plans are to build an IndieAuth provider and management tool/app and integrate that with Authelia for authorising and protecting internal resources. As it turns out, I've already built IndieAuth provider support in all [Yarn.social](https://yarn.social)'s backend `yarnd`, as I'm quite familiar with IndieAuth I _think_ this is what I'd like to do.

As long as you're open to `LDAP`, `YARM file` and `IndieAuth`? -- It _would_ mean that an Authelia instance _could_ in theory "auth" against any website or domain that has a valid IndieAuth provider advertised (including for example my pod https://twtxt.net and many others).

**Reply by onedr0p** [reply]
I'm using glauth with authelia, if you ever circle back to that you can take a look at my home ops repo where all my configuration is open source.

https://github.com/onedr0p/home-ops/blob/80105c4f65f14e12e38d670cfdf1a9afc350f1a8/cluster/apps/authentication/
